use ido to create buffer-list

kkatsuyuki opened the pull request:
Ido can filter default buffer-list.
Emacs workspace package "persp-mode" set a filter for ido and provide
the appropriate workspaces which has only the buffers the user chooses.

kkatsuyuki committed:
use ido to create buffer-list

Ido can filter default buffer-list.
Emacs workspace package "persp-mode" set a filter for ido and provide
the appropriate workspaces which has only the buffers the user chooses.

kkatsuyuki commented:
Also, emacs "Helm" package uses ido to create helm-buffer-list, so I refered to it.

jrosdahl commented:
Thanks for the suggestion!

I would prefer to not tie iflipb to use ido functionality. How about introducing an `iflipb-buffer-list-function` variable, which defaults to `iflipb-buffer-list`? `iflipb-buffer-list` would then just run `(buffer-list (selected-frame))`). `iflipb-ido-buffer-list` could be implemented like you suggested, and then you can `(setq iflipb-buffer-list-function 'iflipb-ido-buffer-list)` to use it. What do you think?

kkatsuyuki committed:
Iflipb works without ido

Added iflipb-buffer-list-function variable which is the function
to be used to create the buffer list.
Its default value is iflipb-buffer-list which works without ido.

kkatsuyuki commented:
I think It's a good idea, so I have applied your suggestion.
